qiskit_nature/second_q/mappers/qubit_mapper.py

```diff
@@ -84,31 +87,30 @@ def mode_based_mapping(
 
         # 0. Some utilities
 
-        times_creation_op = []
-        times_annihilation_op = []
-        times_occupation_number_op = []
-        times_emptiness_number_op = []
-        for paulis in pauli_table:
-            real_part = SparsePauliOp(paulis[0], coeffs=[0.5])
-            imag_part = SparsePauliOp(paulis[1], coeffs=[0.5j])
-
-            # The creation operator is given by 0.5*(X - 1j*Y)
-            creation_op = real_part - imag_part
-            times_creation_op.append(creation_op)
-
-            # The annihilation operator is given by 0.5*(X + 1j*Y)
-            annihilation_op = real_part + imag_part
-            times_annihilation_op.append(annihilation_op)
-
-            # The occupation number operator N is given by `+-`.
-            times_occupation_number_op.append(
-                creation_op.compose(annihilation_op, front=True).simplify()
-            )
+        if not (self.times_creation_op and self.times_annihilation_op \
+        and self.times_occupation_number_op and self.times_emptiness_number_op):
```

Comment:
I suggest that this entire code block gets extracted into a private utility method. E.g.
```python
def _cache_pauli_table(self, pauli_table):
    if not (...):
        ...
```